python/cudf/cudf/core/multiindex.py

```diff
@@ -1302,6 +1302,62 @@ def _poplevels(self, level):
 
         return popped
 
+    @_cudf_nvtx_annotate
+    def swaplevel(self, i=-2, j=-1):
+        """
+        Swap level i with level j.
+        Calling this method does not change the ordering of the values.
+
+        Parameters
+        ----------
+        i : int or str, default -2
+            First level of index to be swapped.
+        j : int or str, default -1
+            Second level of index to be swapped.
+
+        Returns
+        -------
+        MultiIndex
+            A new MultiIndex.
+
+
+        Examples
+        --------
+        >>> import cudf
+        >>> mi = cudf.MultiIndex(levels=[['a', 'b'], ['bb', 'aa']],
+        ...                    codes=[[0, 0, 1, 1], [0, 1, 0, 1]])
+        >>> mi
+        MultiIndex([('a', 'bb'),
+            ('a', 'aa'),
+            ('b', 'bb'),
+            ('b', 'aa')],
+           )
+        >>> mi.swaplevel(0, 1)
+        MultiIndex([('bb', 'a'),
+            ('aa', 'a'),
+            ('bb', 'b'),
+            ('aa', 'b')],
+           )
+        """
+        i = self._level_index_from_level(i)
+        j = self._level_index_from_level(j)
+        name_i = self._data.names[i]
+        name_j = self._data.names[j]
+        new_data = {}
+        for k, v in self._data.items():
+            if k not in (name_i, name_j):
+                new_data[k] = v
+            elif k == name_i:
+                new_data[name_j] = self._data[name_j]
+            elif k == name_j:
+                new_data[name_i] = self._data[name_i]
+        if None in self.names:
```

Comment:
Hmm why should all the result names be `None` if there was at least one name that was `None`

e.g. `self.names == [None, "a"]`, shouldn't `"a"` be kept?

Reply:
Yep, it should be ["a", None] in the swapped index. I will update accordingly.